src/btree/bt_vrfy.c

```diff
@@ -658,16 +658,21 @@ __verify_tree(
             ++vs->depth;
             ret = __wt_page_in(session, child_ref, 0);
 
-            /*
-             * If configured, continue traversing through the pages of the tree even after
-             * encountering errors reading in the page.
-             */
-            if (vs->read_corrupt && ret != 0) {
-                if (vs->verify_err == 0)
-                    vs->verify_err = ret;
-                continue;
-            } else
-                WT_RET(ret);
+            if (ret != 0) {
+                if (vs->dump_address)
+                    WT_TRET(__wt_msg(session, "ERROR unreadable column internal page: %s",
+                      __verify_addr_string(session, ref, vs->tmp1)));
+                /*
+                 * If configured, continue traversing through the pages of the tree even after
+                 * encountering errors reading in the page.
+                 */
+                if (vs->read_corrupt) {
```

Comment:
I'm wondering if this logic might be simplified by "bailing out" earlier:
```suggestion
                if (!vs->read_corrupt)
                     WT_RET(ret);
                if (vs->verify_err == 0)
                     vs->verify_err = ret;
                continue;
```
Also is it important that we retain the first error in `verify_err`? If not: it is possible to simplify too:
```suggestion
                if (!vs->read_corrupt)
                     WT_RET(ret);
                
                vs->verify_err = ret;
                continue;
```

Reply:
Thank you, i will implement the first change. As to  the importance of retaining specifically the first error, the original code done this so i had assumed that the error code of the first page we could not read, holds more value than others? But i am open to discuss changing this to retain the error code of the last page that we couldn't read.

Reply:
fixed https://github.com/wiredtiger/wiredtiger/pull/10095/commits/f85f791c10dd0c8e002842ad987181b795949e29